WordPress/Classes/ViewRelated/Blog/LanguageViewController.swift

```diff
@@ -0,0 +1,140 @@
+            cell?.accessoryType = .DisclosureIndicator
+            WPStyleGuide.configureTableViewCell(cell)
+        }
+        
+        configureTableViewCell(cell!)
+        
+        return cell!
+    }
+    
+    public override func tableView(tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return WPTableViewSectionHeaderFooterView.heightForFooter(footerText, width: view.bounds.width)
+    }
+    
+    public override func tableView(tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        let headerView = WPTableViewSectionHeaderFooterView(reuseIdentifier: nil, style: .Footer)
+        headerView.title = footerText
+        return headerView
+    }
+
+    
+    
+    // MARK: - UITableViewDelegate Methods
+    public override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        pressedLanguageRow()
+    }
+    
+    
+    
+    // MARK: - Private Methods
+    private func configureTableViewCell(cell: UITableViewCell) {
+        let languageId = blog.settings.languageID.integerValue
+        cell.textLabel?.text = NSLocalizedString("Language", comment: "Language of the current blog")
+        cell.detailTextLabel?.text = Languages.sharedInstance.nameForLanguageWithId(languageId)
+    }
+    
+    private func pressedLanguageRow() {
+        // Setup Properties
+        let headers = [
+            NSLocalizedString("Popular languages", comment: "Section title for Popular Languages"),
+            NSLocalizedString("All languages", comment: "Section title for All Languages")
+        ]
+
+        let languages   = Languages.sharedInstance.grouped
+        let titles      = languages.map { $0.map { $0.name } }
+        let subtitles   = languages.map { $0.map { $0.description } }
+        let values      = languages.map { $0.map { $0.languageId } } as [[NSObject]]
+        
+        // Setup ListPickerViewController
+        let listViewController = SettingsListPickerViewController(style: .Grouped)
+        listViewController.title = NSLocalizedString("Site Language", comment: "Title for the Language Picker View")
+        listViewController.selectedValue = blog.settings.languageID
+        listViewController.titles = titles
+        listViewController.subtitles = subtitles
+        listViewController.values = values
+        listViewController.headers = headers
+        listViewController.onChange = { [weak self] (selected: AnyObject) in
+            guard let newLanguageID = selected as? NSNumber else {
+                return
+            }
+            
+            self?.onChange?(newLanguageID)
+        }
+        
+        navigationController?.pushViewController(listViewController, animated: true)
+    }
+    
+    
+
+    // MARK: - Private Constants
+    private let reuseIdentifier = "reuseIdentifier"
+    private let footerText = NSLocalizedString("The language in which this site is primarily written. " +
```

Comment:
This message can be a bit misleading because we don't support changing the interface language on Account Settings yet.